gmail/src/services/log_email.ts

```diff
@@ -1,18 +1,59 @@
 import { postJsonRpc } from "../utils/http";
+import { escapeHtml } from "../utils/html";
 import { URLS } from "../const";
 import { State } from "../models/state";
+import { Email } from "../models/email";
+import { ErrorMessage } from "../models/error_message";
+import { _t } from "../services/translation";
+
+/**
+ * Format the email body before sending it to Odoo.
+ * Add error message at the end of the email, fix some CSS issues,...
+ */
+function _formatEmailBody(email: Email, error: ErrorMessage): string {
+    let body = email.body;
+
+    body = "<span>" + _t("From: ") + escapeHtml(email.contactEmail) + "</span><br/><br/>" + body;
```

Comment:
Not an expert, but isn't it some nicer string formatting than doing concatenation ?